packages/headless/doc-parser/src/interface-resolver.ts

```diff
@@ -146,21 +148,28 @@ function buildEntityFromParamAndResolveTypeAlias(
 ): Entity {
   const entity = buildParamEntity(p);
   const alias = extractTypeName(p.parameterTypeExcerpt);
-  const typeAlias = findApi(entry, alias) as ApiTypeAlias;
+  const typeAlias = findApi(entry, alias.typeName) as ApiTypeAlias;
   const type = typeAlias.typeExcerpt.text;
 
   return {...entity, type};
 }
 
 function buildObjEntityFromParam(entryPoint: ApiEntryPoint, p: Parameter) {
-  const typeName = extractTypeName(p.parameterTypeExcerpt);
+  const {typeName, typeNameWithGenerics} = extractTypeName(
+    p.parameterTypeExcerpt
+  );
   const apiInterface = findApi(entryPoint, typeName) as ApiInterface;
   const members = resolveInterfaceMembers(entryPoint, apiInterface);
   const entity = buildParamEntity(p);
 
-  return buildObjEntity({entity, members, typeName});
+  return buildObjEntity({entity, members, typeName: typeNameWithGenerics});
 }
 
 function extractTypeName(excerpt: Excerpt) {
-  return excerpt.spannedTokens[0].text;
+  const typeName = excerpt.spannedTokens[0].text;
+  const typeNameWithGenerics = excerpt.spannedTokens
```

Comment:
A few questions around this, but some context first:

The object entities have a `type` which is the full type, and a `typeName` which is the name of the interface that we can use to perform a search with. The `typeName` is also used as the name of the extracted type, which I imagine will end up in the url hash.

I see on lines 100 and 165 that the `typeName` is now set to `typeNameWithGenerics`. I feel this could become confusing, since the `typeName now holds two "kinds" of values.

/end of context

1. Could we leverage the `type` field to get the `typeNameWithGenerics`? All we would need to do is replace the square brackets?
2. If deriving the string from `type` is too complex, maybe we could add another property rather than placing two kinds of values in the same property?
3. Would `excerpt.text` return the same value as joining all the tokens below? The final `replace` would still be needed of course. We could also place the logic in it's own function?